File: js/12847.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const [n, m] = inputs[0].split(' ').map(Number);
const pay = inputs[1].split(' ').map(Number);

let currProfit = pay.slice(0, m).reduce((a, b) => a + b, 0);
let answer = currProfit;

for (let i = 0; i < n - m; i++) {
  currProfit -= pay[i];
  currProfit += pay[i + m];
  answer = Math.max(answer, currProfit);
}

console.log(answer);

File: c++/12847.cpp
#include <algorithm>
#include <iostream>

using namespace std;

int pay[100000];

int main() {
  freopen("example.txt", "r", stdin);
  int n, m;
  cin >> n >> m;
  long long currProfit = 0;

  for (int i = 0; i < n; i++) {
    cin >> pay[i];
    if (i < m) currProfit += pay[i];
  }

  long long answer = currProfit;

  for (int i = 0; i < n - m; i++) {
    currProfit -= pay[i];
    currProfit += pay[i + m];
    answer = max(answer, currProfit);
  }

  cout << answer;

  return 0;
}